contrib/ansible/roles/download/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+
+- name: Delete k3s if already present
+  file:
+    path: /usr/local/bin/k3s
+    state: absent
+
+- name: Download k3s binary x64
+  get_url:                                                           
+      url: https://github.com/rancher/k3s/releases/download/{{ k3s_version }}/k3s
+      dest: /usr/local/bin/k3s
+      owner: root
+      group: root
+      mode: 755
+  when: ( ansible_facts.userspace_architecture == "x86_64" )
```

Comment:
I think it can be done by constructing a dictionary of values and combine all these tasks into one:

binary_map = {
"x86_64": "k3s"
"arm_64": "k3s-arm64"
...
}

Reply:
I don't know how to make it work could you pls help me and perhaps make a PR on my repo: https://github.com/itwars/k3s-ansible ?